app/workers/virtual_agent_store_user_info_job.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class VirtualAgentStoreUserInfoJob
+  include Sidekiq::Worker
+
+  def perform(user_info, action_type, _meta)
+    user_record =
+      VirtualAgentUserAccessRecord.create(
+        action_type: action_type, first_name: user_info['first_name'],
+        last_name: user_info['last_name'], ssn: user_info['ssn'], icn: user_info['icn']
```

Comment:
I think here is where you would need to decrypt the `user_info['ssn']` I think some tests may help to verify the expected behavior.

Reply:
Thank you for the valuable feedback. I've added unit testing for the job.

Reply:
Wasn't sure if it was better practice to store the info as encrypted & decrypt at read time, or to simply store as decrypted. Based off your suggestion and the fact that other tables store decrypted data, decided it was ok to decrypt at store time

Reply:
It's also possible that you will want to save the ssn in the db as encrypted as well. This will require a schema change + code on the model level if that is the route that you want to go. Encrypting data is rest is probably best practice